web/pdf_find_controller.js

```diff
@@ -47,47 +47,177 @@ const CHARACTERS_TO_NORMALIZE = {
   "\u00BE": "3/4", // Vulgar fraction three quarters
 };
 
+// These diacritics aren't considered as combining diacritics
+// when searching in a document:
+//   https://searchfox.org/mozilla-central/source/intl/unicharutil/util/is_combining_diacritic.py.
+// The combining class definitions can be found:
+//   https://www.unicode.org/reports/tr44/#Canonical_Combining_Class_Values
+// Category 0 corresponds to [^\p{Mn}].
+const DIACRITICS_EXCEPTION = new Set([
+  // UNICODE_COMBINING_CLASS_KANA_VOICING
+  // https://www.compart.com/fr/unicode/combining/8
+  0x3099, 0x309a,
+  // UNICODE_COMBINING_CLASS_VIRAMA (under 0xFFFF)
+  // https://www.compart.com/fr/unicode/combining/9
+  0x094d, 0x09cd, 0x0a4d, 0x0acd, 0x0b4d, 0x0bcd, 0x0c4d, 0x0ccd, 0x0d3b,
+  0x0d3c, 0x0d4d, 0x0dca, 0x0e3a, 0x0eba, 0x0f84, 0x1039, 0x103a, 0x1714,
+  0x1734, 0x17d2, 0x1a60, 0x1b44, 0x1baa, 0x1bab, 0x1bf2, 0x1bf3, 0x2d7f,
+  0xa806, 0xa82c, 0xa8c4, 0xa953, 0xa9c0, 0xaaf6, 0xabed,
+  // 91
+  // https://www.compart.com/fr/unicode/combining/91
+  0x0c56,
+  // 129
+  // https://www.compart.com/fr/unicode/combining/129
+  0x0f71,
+  // 130
+  // https://www.compart.com/fr/unicode/combining/130
+  0x0f72, 0x0f7a, 0x0f7b, 0x0f7c, 0x0f7d, 0x0f80,
+  // 132
+  // https://www.compart.com/fr/unicode/combining/132
+  0x0f74,
+]);
+const DIACRITICS_EXCEPTION_STR = Array.from(DIACRITICS_EXCEPTION.values())
```

Comment:
Nit: I'm sort-of partial to the slightly shorter `[...DIACRITICS_EXCEPTION]` format.